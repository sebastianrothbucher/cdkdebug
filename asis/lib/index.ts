import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import {ApiConstruct} from "./constructs/api";

export class CommonjsStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);
    new ApiConstruct(this, "api");
  }
}
