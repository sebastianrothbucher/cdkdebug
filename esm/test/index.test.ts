import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';

import { CommonjsStack } from '../lib';

expect.addSnapshotSerializer({
    test: (val) => typeof val === 'string',
    print: (val) =>
        `"${(val as string)
            .replace(/([A-Fa-f0-9]{64})\.(json|zip)|(SsrEdgeFunctionCurrentVersion[A-Fa-f0-9]{40})/, '[FILENAME REMOVED]')
            .replace(
                /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/,
                '[VERSION REMOVED]',
            )}"`,
});

test('Test mvp stack (dev)', async () => {
    const stack = new CommonjsStack(new App(), "stack", { env: { region: 'eu-central-1' } });
    expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();
});
