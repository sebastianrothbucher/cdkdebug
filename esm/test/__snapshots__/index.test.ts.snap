// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test mvp stack (dev) 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "apiNoneDs2753AD49": {
      "Properties": {
        "ApiId": {
          "Fn::GetAtt": [
            "apiSomeAPI4D59EA3E",
            "ApiId",
          ],
        },
        "Name": "NoneDs",
        "Type": "NONE",
      },
      "Type": "AWS::AppSync::DataSource",
    },
    "apiSomeAPI4D59EA3E": {
      "Properties": {
        "AuthenticationType": "API_KEY",
        "Name": "SomeAPI",
      },
      "Type": "AWS::AppSync::GraphQLApi",
    },
    "apiSomeAPIDefaultApiKeyD853C696": {
      "DependsOn": [
        "apiSomeAPISchemaF217F45F",
      ],
      "Properties": {
        "ApiId": {
          "Fn::GetAtt": [
            "apiSomeAPI4D59EA3E",
            "ApiId",
          ],
        },
      },
      "Type": "AWS::AppSync::ApiKey",
    },
    "apiSomeAPIQuerytestres5A053FC2": {
      "DependsOn": [
        "apiNoneDs2753AD49",
        "apiSomeAPISchemaF217F45F",
      ],
      "Properties": {
        "ApiId": {
          "Fn::GetAtt": [
            "apiSomeAPI4D59EA3E",
            "ApiId",
          ],
        },
        "Code": "
        export function request(ctx) {
          return {};
        }
        export function response(ctx) {
          return 42;
        }",
        "DataSourceName": "NoneDs",
        "FieldName": "test",
        "Kind": "UNIT",
        "Runtime": {
          "Name": "APPSYNC_JS",
          "RuntimeVersion": "[VERSION REMOVED]",
        },
        "TypeName": "Query",
      },
      "Type": "AWS::AppSync::Resolver",
    },
    "apiSomeAPISchemaF217F45F": {
      "Properties": {
        "ApiId": {
          "Fn::GetAtt": [
            "apiSomeAPI4D59EA3E",
            "ApiId",
          ],
        },
        "Definition": "
type Query @aws_api_key {
    test: Int
}

schema {
    query: Query
}",
      },
      "Type": "AWS::AppSync::GraphQLSchema",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
